src/subdomains/supporting/payin/services/payin.service.ts

```diff
@@ -96,23 +95,23 @@ export class PayInService {
     await this.payInRepository.save(payIn);
   }
 
-  async updateAmlCheck(payInId: number, amlCheck: CheckStatus): Promise<void> {
-    await this.payInRepository.update(payInId, { amlCheck });
+  async updatePayInAction(payInId: number, amlCheck: CheckStatus): Promise<void> {
+    if (![CheckStatus.FAIL, CheckStatus.PASS].includes(amlCheck)) return;
+
+    await this.payInRepository.update(payInId, {
+      action: amlCheck === CheckStatus.PASS ? PayInAction.FORWARD : PayInAction.WAITING,
+    });
   }
 
-  async returnPayIn(
-    payIn: CryptoInput,
-    purpose: PayInPurpose,
-    returnAddress: BlockchainAddress,
-    route: Staking | Sell | Swap,
-  ): Promise<void> {
-    const amlCheck = route.user.userData.kycLevel === KycLevel.REJECTED ? CheckStatus.FAIL : CheckStatus.PASS;
-    payIn.triggerReturn(purpose, returnAddress, route, amlCheck);
+  async returnPayIn(payIn: CryptoInput, returnAddress: string, chargebackAmount: number): Promise<void> {
+    if (payIn.action === PayInAction.FORWARD) throw new BadRequestException('CryptoInput already forwarded');
+
+    payIn.triggerReturn(BlockchainAddress.create(returnAddress, payIn.asset.blockchain), chargebackAmount);
 
     if (payIn.transaction)
       await this.transactionService.update(payIn.transaction.id, {
         type: TransactionTypeInternal.CRYPTO_INPUT_RETURN,
-        user: route.user,
+        user: payIn.route.user,
       });
```

Comment:
Remove this?

Reply:
I don't know, not quite sure. I thought again about it's technically not completely right but maybe it makes sense to just recognize a cryptoInput return directly, or?